### PR TITLE
Adding permission requirements for encrypted AWS S3 buckets within enterprise-s3-permission.md

### PR DESCRIPTION
**Pull request by melscoop**
### Why:

This permission is a requirement for encrypted buckets on S3 that is currently undocumented.

### Check off the following:

- [ ] I have reviewed my changes in staging (look for the latest deployment event in your pull request's timeline, then click **View deployment**).
- [ ] For content changes, I have completed the [self-review checklist](https://github.com/github/docs/blob/main/CONTRIBUTING.md#self-review).

### Writer impact (This section is for GitHub staff members only):

- [ ] This pull request impacts the contribution experience
  - [ ] I have added the 'writer impact' label
  - [ ] I have added a description and/or a video demo of the changes below (e.g. a "before and after video")

### CODE REVIEW

**Comment by ramyaparimi** [comment]
@melscoop I noticed that this PR was opened in draft mode. Just wanted to check if that was your intention 🐬

**Comment by melscoop** [comment]
> @melscoop I noticed that this PR was opened in draft mode. Just wanted to check if that was your intention 🐬

Yes I mean't to come back and set it to open the next day. I'll do that now.

**Review by verrysaputra — COMMENTED** [review]
More posts by page

File: data/reusables/actions/enterprise-s3-permission.md [hunk 1]
```diff
@@ -7,3 +7,5 @@
 * `s3:AbortMultipartUpload`
 * `s3:DeleteObject`
 * `s3:ListBucket`
+* `kms:GenerateDataKey` (if KMS encryption has been enabled)
```
Comment (skedwards88) [inline data/reusables/actions/enterprise-s3-permission.md:10]:
```suggestion
* `kms:GenerateDataKey` (if Key Management Service (KMS) encryption has been enabled)
```

**Review by skedwards88 — COMMENTED** [review]
✨ Thanks for this addition! I added a small suggestion to define KMS, which I will go ahead an incorporate before getting this merged for you.

**Review by skedwards88 — APPROVED** [review]

**Pull request merged**